great_expectations/data_context/store/database_store_backend.py

```diff
@@ -110,9 +110,10 @@ def __init__(  # noqa: C901 - 16
             table = Table(table_name, meta, *cols)
             try:
                 if self._schema_name:
-                    self.engine.execute(
-                        sa.text(f"CREATE SCHEMA IF NOT EXISTS {self._schema_name};")
-                    )
+                    with self.engine.begin() as connection:
```

Comment:
@Shinnnyshinshin As mentioned in Slack, maybe this pattern will help in some ways?
```
    @contextmanager
    def _get_managed_scoped_db_session(self):
        session = self._scoped_db_session
        try:
            yield session
            session.commit()
        except Exception as e:
            session.rollback()
            raise
        finally:
            session.close()
            session.remove()
```

or this:
```
def _make_query(self, query: str):
  with self.engine.connect() as connection:
    return connection.execute(query)

def some_function(self):
  query = "some query"
  return _make_query(query)
```

Happy to discuss -- thanks!

Reply:
update here : The scope of the changes in this PR don't have a large number of connections that need to be managed (it's only 2 max) so the pattern may not be as helpful at the moment. However, I think it will come up very soon as we tackle subsequent SqlAlchemy 2 issues. Will definitely keep your method and branch in mind.